src/systemctl/systemctl-list-units.c

```diff
@@ -932,3 +932,230 @@ int verb_list_automounts(int argc, char *argv[], void *userdata) {
 
         return r;
 }
+
+struct path_info {
+        const char *machine;
+        const char *id;
+
+        char *path;
+        char *condition;
+
+        /* Note: triggered is a list here, although it almost certainly will always be one
+         * unit. Nevertheless, dbus API allows for multiple values, so let's follow that. */
+        char** triggered;
+
+        /* The strv above is shared. free is set only in the first one. */
+        bool own_triggered;
+};
+
+struct path_infos {
+        size_t count;
+        struct path_info *items;
+};
+
+static int path_info_compare(const struct path_info *a, const struct path_info *b) {
+        int r;
+
+        assert(a);
+        assert(b);
+
+        r = strcasecmp_ptr(a->machine, b->machine);
+        if (r != 0)
+                return r;
+
+        r = strcmp(a->path, b->path);
+
+        r = strcmp(a->condition, b->condition);
+        if (r != 0)
+                return r;
+
+        return strcasecmp_ptr(a->id, b->id);
+}
+
+static void path_infos_free(struct path_infos *ps) {
```

Comment:
This does not free `struct path_infos`, but frees the list. Please rename this `path_infos_done()`.

Reply:
Yeah. I took `sd_bus_error_free` for inspiration, but that name's probably a relic of history that cannot be changed because it's API...